packages/aws-cdk/lib/cli/activity-printer/current.ts

```diff
@@ -0,0 +1,170 @@
+import * as util from 'util';
+import * as chalk from 'chalk';
+import { ActivityPrinterBase, ActivityPrinterProps } from './base';
+import { RewritableBlock } from './display';
+import type { StackActivity } from '../../api/stack-events';
+import { padLeft, padRight, stackEventHasErrorMessage } from '../../util';
+
+/**
+ * Activity Printer which shows the resources currently being updated
+ *
+ * It will continuously re-update the terminal and show only the resources
+ * that are currently being updated, in addition to a progress bar which
+ * shows how far along the deployment is.
+ *
+ * Resources that have failed will always be shown, and will be recapitulated
+ * along with their stack trace when the monitoring ends.
+ *
+ * Resources that failed deployment because they have been cancelled are
+ * not included.
+ */
+export class CurrentActivityPrinter extends ActivityPrinterBase {
+  /**
+   * This looks very disorienting sleeping for 5 seconds. Update quicker.
+   */
+  public readonly updateSleep: number = 2_000;
```

Comment:
As discussed in the call, the producer decides the polling interval, and we're not allowing it to  vary based on the printer just yet. This can go away.

Reply:
done